### PR TITLE
[PIE-2026] Separate in-sync from sync-status listeners

### PR DESCRIPTION
**Pull request by mbaxter**
## PR description
In order to prepare for updates to our `SyncStatus` events, this PR reworks sync-related listener logic so that we have a clean separation between consumers who care whether the node is "in-sync" according to some threshold vs consumers who care about the actual `SyncStatus` updates.

Changes:
* Add the ability to add "in-sync" listeners with arbitrary sync tolerances
* Deduplicate "in-sync" analysis by having `SyncStatusNodePermissioningProvider` subscribe to in-sync events directly
* Rework `SyncStateTest` to use `EthProtocolManagerTestUtil` (less mocking of peer behavior) so that various listeners are set up correctly
* Try to name subscriber methods more consistently
* Simplify `Synchronizer` subscription management by just delegating to `SyncState`

### CODE REVIEW

File: util/src/main/java/org/hyperledger/besu/util/Subscribers.java [hunk 1]
```diff
@@ -152,4 +152,14 @@ public int getSubscriberCount() {
       return 0;
     }
   }
+
+  public interface Unsubscriber {
```
Comment (shemnon) [inline util/src/main/java/org/hyperledger/besu/util/Subscribers.java]: This creates two patterns now, both of which are present in the codebase, a long subscriber ID and a unsubscriber object.  We should pick one.  Perhaps for this release we stick with the existing pattern and in another PR do a full switchover if it is better.  Having multiple subscribe patterns (that are already different from the Java event listener pattern) makes the code less consistant and readable.

**Comment by lucassaldanha** [comment]
I like what this PR is doing and how it enables listeners to specify their tolerance. However, I agree with Danno that we should probably remove the Unsubscriber for now to avoid yet another way to unsubscribe. I'd rather if we stick with the subscription id for now. Unless you have a good reason for introducing the Unsubscriber right now.

File: util/src/main/java/org/hyperledger/besu/util/Subscribers.java (hunk 1 above)
Comment (mbaxter) [reply util/src/main/java/org/hyperledger/besu/util/Subscribers.java]: Good points, I can cut this.  I had a more complicated set-up during development that made tracking subscribers by ids complex, which is why I originally introduced this.  But there's no need for it at this point.

Comment (mbaxter) [reply util/src/main/java/org/hyperledger/besu/util/Subscribers.java]: Removed

**Pull request merged**